### ISSUE TITLE
Application terminates without downloading anything

### ISSUE THREAD

**Issue by BlackB846** [TheAxeDude/ZiniGo#26]
Hi,

up until recently, the program worked fine. I tried to download the newest issues today and the app terminated without doing anything ("Terminating the application...").

Any idea if Zinio changed their web service?

Thanks

**Comment by BlackB846** [comment]
Seems to be working again - no idea if it was an issue on their web site.

**Comment by jmskuo** [comment]
Hi,
I had same problem when i run the app terminated without doing anything.

Config file loaded
Username taken from config file
password taken from config file
chromepath taken from config file
Terminating the application...

**Comment by BlackB846** [comment]
I can confirm @jmskuo - the problem is back. I've tried for several days but the app terminates without downloading anything nor displaying the list of downloaded magazines.

**Comment by SR-G** [comment]
Same problem here... Indeed, one month ago everything was working fine...

It seems it's a "login" issue : 
- all steps are executed but no library are found
- when dumping the URL used and the response, it seems that usertoken is empty when trying to retrieve the list of libraries (second step) (i.e., in "GetLibrary", line `"https://zinio.com/api/newsstand/newsstands/101/users/"+userToken.Data.User.UserIDString+"/library_issues?limit="+strconv.Itoa(issuesToFetch)+"&page="+strconv.Itoa(pageToFetch)`, the userToken.Data.User.UserIDString stuff is empty : response is `{"status":false,"error":{"message":"Invalid Json Web Token in AsyncMiddleware","internal_code":"INVALID_TOKEN"}}`
- about login (first step), the URL triggered is `https://www.zinio.com/api/login` : 
  - for now it seems to render a `{"status":false,"error":{"message":"Route not found in ApiRouter","internal_code":"CONTENT_ERROR"}}` response (when the URL is used in a browser)
  - more interestantly, when executing the Zinigo code, i get this response `{"status":false,"error":{"message":"There have been validation errors: Invalid value: device = undefined,Invalid value: newsstand = undefined","internal_code":"INVALID_PARAMS"}}`

But i'm not able to go further / i don't know where the ZINIO API is described.

**Comment by hieronymousch** [comment]
Can confirm same issue on both API endpoints

**Comment by SR-G** [comment]
If i add a "device" entry in the AUTH JSON token, i then get a "Invalud value: newsstand = undefined" error (instead of the error about "device").

But then i'm not able to add a "newsstand" (= something) value, i always get a "Incorrect JSON structure" error (if i remember correctly) (maybe it's a whole object that is expected and not a value).

Anyway not able to progress further + it seems these login API are not used by the official web client itself

**Comment by TheBanana926** [comment]
I am still having this issue / haven't been able to get the software started yet on my Windows 11 PC- the best I have been able to do is have the console run when opening Ziniogo exe... and then it randomly closes. Have found the readme confusing slightly. Any help?

**Comment by elton667** [comment]
I had the same problem.....

**Comment by OnoBeachBum** [comment]
Any updates on this, I have the same problem?

**Comment by TheAxeDude** [comment]
I'll try take a look into these issues - it seems they've added a few new fields into their API.

**Comment by hieronymousch** [comment]
Thanks for you time and happy to support with some coffee or patreon!

**Comment by TheAxeDude** [comment]
Hi all, newly built binaries here: https://github.com/TheAxeDude/ZiniGo/tree/master/built - please let me know if they work okay? I havent been able to test on any platform except windows.

**Comment by SR-G** [comment]
Tested just righ now on linux (archlinux), hence "Linux_AMD64" binary : everything is working fine now.

Thanks for this update !

**Comment by hieronymousch** [comment]
Same here, thanks a lot for this!!

**Comment by OnoBeachBum** [comment]
Works great for me, Thanks!

**Comment by BlackB846** [comment]
Download works great using the 64bit Windows version on a Windows 11 machine. 

I'm getting a sort of degub dump on the screen when the app starts up - possibly a list of data it retrieves incl.  a "Fetching page: xxx" every now and then. Once that dump is over, the remaining output is the same as before.

**Comment by elton667** [comment]
windows 11 OK
Thanks for this update !

**Comment by jmskuo** [comment]
I still got {"status":false,"error":{"message":"Invalid Json Web Token in AsyncMiddleware","internal_code":"INVALID_TOKEN"}} error message on windows 11.

**Comment by OnoBeachBum** [comment]
I tried the x64 version on both Windows 10(21H2) and Windows 11 (22H2) and it works fine. No problems.

**Comment by gabag30** [comment]
seems to be working pretty well in Apple M1.  Thanks a lot
cheers

**Comment by lvthunder** [comment]
I'm getting this.

`GettingLogin
GotLogin
Fetching Library
Fetching page:1
{"status":false,"error":{"message":"Invalid Json Web Token in AsyncMiddleware","internal_code":"INVALID_TOKEN"}}
Terminating the application...`

**Comment by Neon22** [comment]
Getting same response:
GettingLogin
GotLogin
Fetching Library
Fetching page:1
{"status":false,"error":{"message":"Invalid Json Web Token in AsyncMiddleware","internal_code":"INVALID_TOKEN"}}
Terminating the application...

**Comment by TheAxeDude** [comment]
Are you still getting this issue? Just tried it on my side with no issue. Also, did it work before?

**Comment by lvthunder** [comment]
Yes, I'm still getting it.  When I posted was the first time I've tried using this.

**Comment by Neon22** [comment]
Interestingly I get the same output if I deliberately use a wrong password. Its possible some other thing is wrong with my input (using the config file) that the feedback is indicating is right but is wrong.. Hmmm.
